FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt

```diff
@@ -256,6 +268,324 @@ object PgpMsg {
+    val name = att.fileName ?: ""
+    val type = att.contentType
+    val length = att.size
+    if (hiddenFileNames.contains(name)) {
+      // PGPexch.htm.pgp is html alternative of textual body content produced
+      // by the PGP Desktop and GPG4o
+      return TreatAs.HIDDEN
+    } else if (name == "signature.asc" || type == "application/pgp-signature") {
+      return TreatAs.SIGNATURE
+    } else if (name == "" && !type.startsWith("image/")) {
+      return if (length < 100)  TreatAs.SIGNATURE else TreatAs.ENCRYPTED_MSG
+    } else if (name == "msg.asc" && length < 100 && type == "application/pgp-encrypted") {
+      // mail.ch does this - although it looks like encrypted msg,
+      // it will just contain PGP version eg "Version: 1"
+      return TreatAs.SIGNATURE
+    } else if (encryptedMsgNames.contains(name)) {
+      return TreatAs.ENCRYPTED_MSG
+    } else if (encryptedFileRegex.containsMatchIn(name)) {
+      // ends with one of .gpg, .pgp, .???.asc, .????.asc
+      return TreatAs.ENCRYPTED_FILE
+    } else if (privateKeyRegex.containsMatchIn(name)) {
+      return TreatAs.PRIVATE_KEY
+    } else if (type == "application/pgp-keys") {
+      return TreatAs.PUBLIC_KEY
+    } else if (publicKeyRegex1.containsMatchIn(name)) {
+      // name starts with a key id
+      return TreatAs.PUBLIC_KEY
+    } else if (
+      name.toLowerCase(Locale.ROOT).contains("public") &&
+      publicKeyRegex2.containsMatchIn(name)
+    ) {
+      // name contains the word "public", any key id and ends with .asc
+      return TreatAs.PUBLIC_KEY
+    } else if (name.endsWith(".asc") && length > 0 && checkForPublicKeyBlock(att.inputStream)) {
+      return TreatAs.PUBLIC_KEY
+    } else if (name.endsWith(".asc") && length < 100000 && !att.isInline()) {
+      return TreatAs.ENCRYPTED_MSG
+    } else {
+      return TreatAs.PLAIN_FILE
+    }
+  }
+
+  @JvmStatic
+  private fun checkForPublicKeyBlock(stream: InputStream): Boolean {
+    val a = ByteArray(100)
+    val r = stream.read(a)
+    if (r < 1) return false
+    val s = String(if (r == a.size) a else a.copyOf(r), StandardCharsets.US_ASCII)
+    return s.contains("-----BEGIN PGP PUBLIC KEY BLOCK-----")
+  }
+
+  private val hiddenFileNames = setOf(
+    "PGPexch.htm.pgp",
+    "PGPMIME version identification",
+    "Version.txt",
+    "PGPMIME Versions Identification"
+  )
+
+  private val encryptedMsgNames = setOf(
+    "message", "msg.asc", "message.asc", "encrypted.asc", "encrypted.eml.pgp",
+    "Message.pgp", "openpgp-encrypted-message.asc"
+  )
+
+  private val encryptedFileRegex = Regex("(\\.pgp\$)|(\\.gpg\$)|(\\.[a-zA-Z0-9]{3,4}\\.asc\$)")
+  private val privateKeyRegex = Regex("(cryptup|flowcrypt)-backup-[a-z0-9]+\\.(key|asc)\$")
+  private val publicKeyRegex1 = Regex("^(0|0x)?[A-F0-9]{8}([A-F0-9]{8})?.*\\.asc\$")
+  private val publicKeyRegex2 = Regex("[A-F0-9]{8}.*\\.asc\$")
 }
+
 //                   || ex.message?.contains("Invalid Curve25519 public key") == true
```

Comment:
Please remove it or describe why we should leave it as-is

Reply:
will remove this